flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/resources/FlinkService.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.kubeclient.resources;
+
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ConfigOptions;
+import org.apache.flink.configuration.Configuration;
+
+import io.fabric8.kubernetes.api.model.Service;
+
+/**
+ * Represent Service resource in kubernetes.
+ */
+public class FlinkService extends Resource<Service> {
+
+	public static final ConfigOption<String> SERVICE_ID = ConfigOptions
```

Comment:
Why not putting this config in `KubernetesConfigOptions`?
You can annotate it with `@Documentation.ExcludeFromDocumentation` if you want to hide it from users.

Reply:
I will move all the internal config options to `KubernetesConfigOptionsInternal`. It will be marked as `@Internal`.

Reply:
Is is true that we never expect users to set such configs?
If so, I'm not sure whether it's a good idea to make them `ConfigOption`s.
Would it work to make them class properties or function params?

Reply:
The config options in `KubernetesConfigOptionsInternal` are only for internal use. It will set to configuration by `KubernetesClusterDescriptor` or some other classes in deploying process. Then the `FlinkKubeClient` only need a updated `Configuration` to construct a new object. Otherwise, we need more arguments for `FlinkKubeClient`.

This way is also used in yarn deployment.

Reply:
OK.